Japanese translation and move original file (#7)
univapay/univapay-php-sdk: README_en.md

# UnivaPay PHP SDK

![Github Actions CI](https://github.com/univapay/univapay-php-sdk/workflows/PHP%20lint%20&%20test/badge.svg)

> This PHP SDK provides a convenient way to integrate your services with the UnivaPay payments gateway.

Reviewer: ```suggestion
This PHP SDK provides a convenient way to integrate your services with the UnivaPay payments gateway.

*[日本語](README.md)*
```

_By default, if the value is not specified during creation, it will be snake-cased from the name_

```php
use UnivapayEnumsChargeStatus;

$values = ChargeStatus::findValues(); // Get a list of all names and values in the enumerator
$chargeStatus = ChargeStatus::PENDING(); // Note the braces at the end
$chargeStatus->getValue() === 'pending'; // true
$chargeStatus === ChargeStatus::fromValue('pending'); // true
// Also works for switch statements
switch ($chargeStatus) {
    case ChargeStatus::PENDING():
        // Do something
        break;
    // ...
}
```

### Updating resource models
To update/refresh any resource models (model classes that extends `Resource`)

```php
$charge->fetch();
```

### Long polling
The following resources supports long polling to wait for the next status change:
- `Charge`
- `Refund`
- `Cancel`
- `Subscription`

> This is useful since these requests initially returns with a `PENDING` status. Long polling allows you to fetch the updated model when the resource has changed its status. If no changes occurs within 5 seconds, it will return the resource at that state.

Reviewer: ```suggestion
This is useful since these requests initially returns with a `PENDING` status. Long polling allows you to fetch the updated model when the resource has changed its status. If no changes occurs within 3 seconds, it will return the resource at that state.
```

```php
use UnivapayRequestsHandlersBasicRetryHandler;

$subscriptionTokenRetryHandler = new BasicRetryHandler(
    UnivapayResourceConflictError::class,
    5, // Tries 5 times
    2, // At 2 seconds interval
    // More specific filtering based on the error, takes in the error as the first parameter
    // return true to retry, false to ignore.
    function (UnivapayResourceConflictError $error) {
        return $error->code === 'NON_UNIQUE_ACTIVE_TOKEN';
    }
);
$client->addHandlers($subscriptionTokenRetryHandler);

// To reset or to clear and add new handlers from scratch
// The rateLimitHandler will be automatically added from UnivapayClientOptions
$client->setHandlers($subscriptionTokenRetryHandler);
```

## SDK Development

Building:
```shell
composer install
npm install

# Optionally
npm install -g grunt
```

Code formatting:
```shell
grunt phpcs
```

Tests:

The following env vars are required when running the tests:

- `UNIVAPAY_PHP_TEST_TOKEN` - This should be a `test` mode token
- `UNIVAPAY_PHP_TEST_SECRET`
- `UNIVAPAY_PHP_TEST_ENDPOINT` - This would point to a local API instance or a staging instance

```shell
grunt phpunit
```

> _Note: CircleCI only runs on branches that has an open PR_

Reviewer: ```suggestion
_Note: Github Actions only runs on branches that has an open PR_
```
